platform/admin/core/admin-core-alertservice-api/src/main/java/org/codice/ddf/admin/core/alert/service/api/internal/AlertsSupplier.java

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.admin.core.alert.service.api.internal;
+
+import java.util.List;
+
+import javax.annotation.Nonnull;
+
+/**
+ * <p>
+ * <b> This code is experimental. While this interface is functional and tested, it may change or be
+ * removed in a future version of the library.</b>
+ * </p>
+ * <p>
+ * While some {@link Alert}s will be pushed to the {@link AlertService}, other {@link Alert}s are
+ * pulled from {@link AlertsSupplier}s by the {@link AlertService}. The {@link #getAlerts()} method
+ * returns the {@link AlertsSupplier}'s list of {@link Alert}s.
+ */
+public interface AlertsSupplier {
```

Comment:
I feel like `AlertSupplier` is more idiomatic than `AlertsSupplier`. We typically only use the plural when a method return a collection. In class names, we almost never use a plural.